dex/farm/src/base_functions.rs

```diff
@@ -99,25 +104,56 @@ pub trait BaseFunctionsModule:
                 &base_exit_farm_result.storage_cache.reward_token_id,
             );
         }
-
         (farming_token_payment, reward_payment).into()
     }
 
+    fn merge_farm_tokens_with_energy(&self, caller: &ManagedAddress) -> EsdtTokenPayment<Self::Api> {
+        let mut payments = self.get_non_empty_payments();
+        let first_payment_nonce = self.clear_payments_claim_progress(caller, &payments);
+        let first_payment = self.pop_first_payment(&mut payments);
+
+        let token_mapper = self.farm_token();
+        let mut output_attributes: FarmTokenAttributes<Self::Api> =
+            self.get_attributes_as_part_of_fixed_supply(&first_payment, &token_mapper);
+        token_mapper.nft_burn(first_payment.token_nonce, &first_payment.amount);
+
+        for payment in &payments {
+            let attributes: FarmTokenAttributes<Self::Api> =
+                self.get_attributes_as_part_of_fixed_supply(&payment, &token_mapper);
+            output_attributes.merge_with(attributes);
+        }
```

Comment:
Seeing as this code is pretty common, rewrite it as method in the utils module, that receives a `PaymentsVec` as argument, and returns `T`, where `T` is the attributes type. Similar to the current `merge_from_payments` implementation, but without any `base_attributes` argument.

I'd say rename the current `merge_from_payments` to `merge_attributes_with_payments`, and name the new one `merge_from_payments`.

Reply:
Done.